mypy/checker.py

```diff
@@ -8579,7 +8527,10 @@ def and_conditional_maps(m1: TypeMap, m2: TypeMap, use_meet: bool = False) -> Ty
         for n1 in m1:
             for n2 in m2:
                 if literal_hash(n1) == literal_hash(n2):
-                    result[n1] = meet_types(m1[n1], m2[n2])
+                    meet_result = meet_types(m1[n1], m2[n2])
+                    if isinstance(meet_result, UninhabitedType):
```

Comment:
This was a preexisting bug that gets exposed. We now use this logic for chained comparisons, in place of the bespoke logic previously